docs/sources/v0.51.x/using-k6-browser/migrating-to-k6-v0-52.md

```diff
@@ -0,0 +1,215 @@
+    await page.screenshot({ path: 'screenshot.png' });
+  } finally {
+    await page.close();
+  }
+}
+```
+
+{{< /code >}}
+
+### Can I promise chain?
+
+If you want to use promise chaining the newly migrated async APIs, You won't be able to do that until the migration. For example, you can't do this yet:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  await page.$('#text1').then((text) => text.type('hello world'));
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Instead you could do this until k6 `v0.52` is released:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const text = await page.$('#text1');
+  await text.type('hello world');
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Working with k6 check
+
+The k6 `check` API wasn't designed to work with async APIs. However there is a workaround. In your current scripts, you may use a `check` like so:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  check(page, {
+    header: (p) => p.locator('h2').textContent() == 'Welcome, admin!',
+  });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Since the `locator.textContent` API is being migrated to async, and `check` doesn't work with async APIs, you will need to do something like this:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const h2 = page.locator('h2');
+  const headerOK = (await h2.textContent()) == 'Welcome, admin!';
+  check(headerOK, { header: headerOK });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Groups
+
+A note on groups, they do not work with async APIs either, there is no workaround as of yet. Here's the [github issue](https://github.com/grafana/k6/issues/2728) that you can follow to keep up to date with relevant news on a groups API that works with async APIs.
+
+# Are there any other upcoming major breaking changes?
+
+We think that this is likely the last large breaking change until the browser module is deemed to be stable enough to be a non-experimental module in k6. When the browser module is no longer experimental, we think it should only be a small change in the import path from `k6/experimental/browser` to `k6/browser`. We have an open [issue on github](https://github.com/grafana/xk6-browser/issues/1117) which you can view and track to see the progress of graduating the browser module out of experimental. For more information on graduating extension out of experimental, refer to the [extension graduation process]({{< relref "../extensions/explanations/extension-graduation" >}}).
```

Comment:
I wouldn't link to the non-experimental issue just yet because most tasks in that issue have not been done yet, and it also explains that the module is not stable, etc. Although we're at an experimental stage, I wouldn't directly advertise it like so.

Reply:
Hmm, yep, I see what you are saying, and it's not particularly user friendly. I was trying to emphasis that this was seriously on the cards and people should be ready for another possible breaking change. At the same time, this could dissuade users from even trying to work with k6 browser if they saw this.

Reply:
Removed in 4978eebbffe68ccbb165f1978df3ccdd4402f825